### PR TITLE
[ci] Clean up ci/ directory (refactor ci/travis)

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Clean up the `ci/` directory. This means getting rid of the `travis/` path completely and moving the files into sensible subdirectories.

Details:
- Moves everything under `ci/travis` into subdirectories, e.g. `ci/build`, `ci/lint`, etc.
- Minor adjustments to some scripts (variable renames)
- Removes the outdated (unused) asan tests

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by krfricke** [comment]
CI seems to be passing fine, tagging @simon-mo for review and a few other folks for code-owner approval.

File: scripts/format.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+../ci/lint/format.sh
```
Comment (krfricke) [inline scripts/format.sh:1]: This is the only script we usaully reference (in PRs)

**Review by ericl — APPROVED** [review]

**Review by gjoliver — APPROVED** [review]
as long as RL tests run.

**Review by simon-mo — APPROVED** [review]
stamp

**Comment by simon-mo** [comment]
can we please keep ./scripts symlink? it's very useful. linking it to lint works

**Comment by krfricke** [comment]
`scripts/format.sh` was kept in this PR before, but I've updated the PR to link to add links to all scripts for backwards compatibility and not to break any existing worklflows.

**Review by amogkam — APPROVED** [review]

**Comment by kouroshHakha** [comment]
Documentation regarding the installation needs to be updated! (for bazel installation)

**Pull request merged**